var/spack/repos/builtin/packages/m4/package.py

```diff
@@ -61,3 +66,25 @@ def configure_args(self):
             args.append('ac_cv_type_struct_sched_param=yes')
 
         return args
+
+    def test(self):
+        m4 = which('m4')
+        assert m4 is not None
+
+        tty.msg('test: Ensuring use of the installed executable')
+        m4_dir = os.path.dirname(m4.path)
+        assert m4_dir == self.prefix.bin
+
+        tty.msg('test: Checking version')
+        output = m4('--version', output=str.split, error=str.split)
+        version_regex = re.compile(r'm4(.+){0}'.format(self.spec.version))
+        assert version_regex.search(output)
+
+        tty.msg('test: Ensuring m4 runs')
+        currdir = os.getcwd()
+        hello_fn = os.path.join(currdir, 'data', 'hello.m4')
+        output = m4(hello_fn, output=str.split, err=str.split)
+        expected_fn = os.path.join(currdir, 'data', 'hello.out')
+        with open(expected_fn) as fd:
+            expected = fd.readlines()
```

Comment:
Use `fd.read()` so you don't have to join them

Reply:
Ok